projects/sandbox/train/train/validation.py

```diff
@@ -245,299 +50,220 @@ class Recorder:
-    background = background[:, :stop]
-    background = torch.Tensor(background)[None, :, None]
-
-    # fold out into windows up front
-    background = torch.nn.functional.unfold(
-        background, (1, num_kernels), dilation=(1, stride_size)
-    )
-
-    # some reshape magic having to do with how the
-    # unfold op orders things. Don't worry about it
-    background = background.reshape(num_ifos, num_kernels, kernel_size)
-    background = background.transpose(1, 0)
-    return background
-
-
-def make_glitches(
-    glitches: Sequence[np.ndarray],
-    background: torch.Tensor,
-    glitch_frac: float,
-) -> torch.Tensor:
-    if len(glitches) != background.size(1):
-        raise ValueError(
-            "Number of glitch tensors {} doesn't match number "
-            "of interferometers {}".format(len(glitches), background.size(1))
-        )
-
-    h1_glitches, l1_glitches = map(torch.Tensor, glitches)
-    num_h1, num_l1 = len(h1_glitches), len(l1_glitches)
-    num_glitches = num_h1 + num_l1
-    num_coinc = int(glitch_frac**2 * num_glitches / (1 + glitch_frac**2))
-    if num_coinc > min(num_h1, num_l1):
-        raise ValueError(
-            f"There are more coincident glitches ({num_coinc}) that there "
-            "are glitches in one of the ifo glitch datasets. Hanford: "
-            "{num_h1}, Livingston: {num_l1}"
-        )
 
-    h1_coinc, h1_glitches = split(h1_glitches, num_coinc / num_h1, 0)
-    l1_coinc, l1_glitches = split(l1_glitches, num_coinc / num_l1, 0)
-    coinc = torch.stack([h1_coinc, l1_coinc], axis=1)
-    num_h1, num_l1 = len(h1_glitches), len(l1_glitches)
-    num_glitches = num_h1 + num_l1 + num_coinc
-
-    # if we need to create duplicates of some of our
-    # background to make this work, figure out how many
-    background = repeat(background, num_glitches)
-
-    # now insert the glitches
-    kernel_size = background.size(2)
-    start = h1_glitches.shape[-1] // 2 - kernel_size // 2
-    slc = slice(start, start + kernel_size)
+@dataclass
+class Validator:
+    tracker: LocalTracker
+    background: torch.Tensor
+    waveforms: torch.Tensor
+    sample_rate: float
+    stride: float
+    injection_stride: float
+    kernel_length: float
+    batch_size: int
+    pool_length: float
+    integration_length: float
+    livetime: float
+    shift: float
+    max_fpr: float
+    device: str
 
-    background[:num_h1, 0] = h1_glitches[:, slc]
-    background[num_h1:-num_coinc, 1] = l1_glitches[:, slc]
-    background[-num_coinc:] = coinc[:, :, slc]
+    def __post_init__(self):
+        self.auroc = BinaryAUROC(max_fpr=self.max_fpr)
+        self.duration = self.background.shape[-1] / self.sample_rate
+        self.kernel_size = int(self.kernel_length * self.sample_rate)
+        self.stride_size = int(self.stride * self.sample_rate)
+        self.pool_size = int(self.pool_length / self.stride)
+
+        integration_size = int(self.integration_length * self.sample_rate)
```

Comment:
This should be `int(self.integration_length * inference_sampling_rate)` (where `inference_sampling_rate = (1 / stride)` right?

Reply:
Oh wow yeah great catch